qiskit/tools/visualization/_state_visualization.py

```diff
@@ -420,56 +511,46 @@ def plot_state_qsphere(rho, filename=None, show=False):
             ax.plot([0], [0], [0], markerfacecolor=(.5, .5, .5),
                     markeredgecolor=(.5, .5, .5), marker='o', markersize=10,
                     alpha=1)
-            if filename:
-                plt.savefig(filename)
-            elif show:
-                plt.show()
             we[prob_location] = 0
         else:
             break
-    return plt.gcf()
+    plt.tight_layout()
+    plt.close(fig)
+    return fig
 
 
-def plot_state(quantum_state, method='city', filename=None, show=False):
+def plot_state(quantum_state, method='city', figsize=None):
     """Plot the quantum state.
 
     Args:
         quantum_state (ndarray): statevector or density matrix
                                  representation of a quantum state.
         method (str): Plotting method to use.
-        filename (str): the output file to save the plot as. If specified it
-            will save and exit and not open up the plot in a new window. If
-            `bloch` method is used a `-n` will be added to the filename before
-            the extension for each qubit.
-        show (bool): If set to true the rendered image will open in a new
-            window
+        figsize (tuple): Figure size in inches,
+
     Returns:
          matplotlib.Figure: The matplotlib.Figure of the visualization
     Raises:
+        ImportError: Requires matplotlib.
         VisualizationError: if the input is not a statevector or density
         matrix, or if the state is not an multi-qubit quantum state.
     """
-
+    if not HAS_MATPLOTLIB:
+        raise ImportError('Must have Matplotlib installed.')
+    warnings.warn("'plot_state' will be depreciated in terra-0.9.  \
+                  Call the desired plotting routine directly.",
+                  DeprecationWarning)
```

Comment:
I would slightly reword this to say something like:

plot_state() is deprecated and will be removed in the 0.9 release.  Instead you can use the plot_state_*() function directly for method being used.

Reply:
The `plot_bloch_vector` does not follow this pattern, nor does `plot_wigner_function`  Rather than have a lengthy depreciation warning, the user can turn to the tutorials or docs for help.

Reply:
Well plot_bloch_vector is a different function that was already part of the api, and has different inputs. That was actually a comment further down about how we lose the functionality to go from rho to multiple bloch vectors after we remove `plot_state()`. As for plot_wigner_function() I'm actually thinking more that we should remove it or at least keep it separate from plot_state() as a solution to #1106 and #1302 .

Part of my issue with the wording here is that it assumes people know what their desired plotting routine is, which they may not if they've only ever used `plot_state()` before. Especially since we don't have anything in the changelog about this yet. If ~50 more characters is too long:

plot_state is deprecated and will be removed in the 0.9 release. Use the plot_state_* functions instead.

Which is only an additional 5.